frontend/src/store/wallet.ts

```diff
@@ -129,6 +129,13 @@ export default function useWalletStore() {
 
   watch(lastWallet, async () => {
     if (lastWallet.value && !userAddress.value) {
+      if (lastWallet.value === 'MetaMask') {
```

Comment:
Does `lastWallet` have a property we can key off of to determine if its injected web3? That way instead of special casing MetaMask we also cover Brave, Frame, Tally, etc.?

But presumably those have different ways of checking if they're unlocked, so we probably need to special case each. In which case, let's leave this as-is and we can special case others as requests for it come in

Reply:
Seems like no based on the fields `WalletState` has (since `our `connectedWallet` value is of that type)

Reply:
Yea, I looked there wasn't another way. It would definitely be nice if they had a generic implementation for this.

Reply:
Cool, this LGTM then!